### PR TITLE
Port to CMake and get the example running again

### PR DESCRIPTION
**Pull request by bawilless**
I've made changes and tested the changes on the Seeed RDB, and both Avnet Starter Kits Rev1 and Rev2.  Please accept this PR so others can benefit from your work and mine!  Thanks,

Brian

### CODE REVIEW

**Comment by xiongyu0523** [comment]
Thanks, will review and test soon

File: arducam_driver/ll.c [hunk 1]
```diff
@@ -5,8 +5,7 @@
 #include <errno.h>
 #include <string.h>
 
-#include "applibs_versions.h"
-
+#include "../applibs_versions.h"
```
Comment (xiongyu0523) [inline arducam_driver/ll.c:8]: What is this for? The current path of project should be the one where CMakeLists.txt located, right? Then why you want '../'?

File: arducam_driver/ArduCAM.c [hunk 2]
```diff
@@ -9,7 +9,7 @@
 #include <stdint.h>
 #include <stdbool.h>
 
-#include "delay.h"
+#include "../delay.h"
```
Comment (xiongyu0523) [inline arducam_driver/ArduCAM.c:12]: Samae as '../applibs._versions.h'

File: CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+azsphere_configure_tools(TOOLS_REVISION "20.10")
+azsphere_configure_api(TARGET_API_SET "7") 
+ 
+add_executable(${PROJECT_NAME} 
+                main.c 
+                arducam_driver/ArduCAM.c 
+                arducam_driver/ll.c 
+                arducam_driver/ov2640_regs.c 
+                delay.c)
+target_include_directories(${PROJECT_NAME} PUBLIC ${AZURE_SPHERE_API_SET_DIR}/usr/include/azureiot ${AZURE_SPHERE_API_SET_DIR}/usr/include/azure_prov_client)
```
Comment (xiongyu0523) [inline CMakeLists.txt:17]: No required for minimum configuration

File: CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+# Set the defines for the build
+target_compile_definitions(${PROJECT_NAME} PUBLIC AZURE_IOT_HUB_CONFIGURED)
```
Comment (xiongyu0523) [inline CMakeLists.txt:20]: No required for minimum configuration

File: CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,45 @@
+#  Copyright (c) Microsoft Corporation. All rights reserved.
+#  Licensed under the MIT License.
+
+cmake_minimum_required(VERSION 3.10)
+
+project(AvnetAzureIoT C)
```
Comment (xiongyu0523) [inline CMakeLists.txt:6]: would you please reset the name of project to azure-sphere-arducam-mini-2mp-plus

File: CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+# Set the defines for the build
+target_compile_definitions(${PROJECT_NAME} PUBLIC AZURE_IOT_HUB_CONFIGURED)
+target_compile_definitions(${PROJECT_NAME} PUBLIC AzureSphere_CA7)
+target_compile_definitions(${PROJECT_NAME} PUBLIC CFG_MODE_JPEG)
+
+target_link_libraries(${PROJECT_NAME} azureiot applibs pthread gcc_s c curl)
+
+azsphere_target_hardware_definition(${PROJECT_NAME} TARGET_DIRECTORY "Hardware/avnet_mt3620_sk_rev2" TARGET_DEFINITION "sample_hardware.json")
+
```
Comment (xiongyu0523) [inline CMakeLists.txt:27]: No required for minimum configuration

File: app_manifest.json [hunk 7]
```diff
@@ -5,7 +5,7 @@
   "EntryPoint": "/bin/app",
   "CmdArgs": [],
   "Capabilities": {
-    "AllowedConnections": [ "<your-storage-account>.blob.core.windows.net" ],
+    "AllowedConnections": [ "bwstorageaccount.blob.core.windows.net" ],
```
Comment (xiongyu0523) [inline app_manifest.json:8]: make this more generic, revert change required

**Review by xiongyu0523 — COMMENTED** [review]
Please review comments, thanks

**Comment by xiongyu0523** [comment]
Merged

File: arducam_driver/ll.c (hunk 1 above)
Comment (xiongyu0523) [reply arducam_driver/ll.c:8]: No problem, it is under a folder

File: arducam_driver/ArduCAM.c (hunk 2 above)
Comment (xiongyu0523) [reply arducam_driver/ArduCAM.c:12]: No problem, it is under a folder

File: CMakeLists.txt (hunk 3 above)
Comment (xiongyu0523) [reply CMakeLists.txt:17]: will be removed later

File: CMakeLists.txt (hunk 4 above)
Comment (xiongyu0523) [reply CMakeLists.txt:20]: will be removed later

File: CMakeLists.txt (hunk 5 above)
Comment (xiongyu0523) [reply CMakeLists.txt:6]: will be corrected later

File: CMakeLists.txt (hunk 6 above)
Comment (xiongyu0523) [reply CMakeLists.txt:27]: change to mt3620_rdb and remove not used lib

**Comment by bawilless** [comment]
> Merged

Thanks for merging in my changes, sorry my CMakeList.txt file was so messy.  I noted that you added the CA certificate and corresponding calls.  I struggled to get that working, thank you!

Do you have any experience sending up very large image files (>200K) by sending multiple chunks of the file?  Since the MT3620 only allows for high level applications to use 256K of memory this is something I need to solve.

Brian

**Pull request merged**